gammapy/datasets/spectrum.py

```diff
@@ -325,15 +327,28 @@ def read(cls, filename, format="ogip", checksum=False, name=None, **kwargs):
         """
         from .io import OGIPDatasetReader
 
-        if format == "gadf":
-            return super().read(
-                filename, format="gadf", checksum=checksum, name=name, **kwargs
-            )
+        if format is None:
+            with fits.open(filename) as hdulist:
+                # Check if the file is empty
+                if len(hdulist) == 0:
+                    raise ValueError(f"File {filename} is empty")
+                # Check for extensions in OGIP format
+                if "SPECTRUM" in hdulist and "OGIP" in hdulist[1].header["HDUCLASS"]:
```

Comment:
```suggestion
                if "SPECTRUM" in hdulist and "OGIP" in hdulist["SPECTRUM"].header["HDUCLASS"]:
```
It seems safer to test this specific hdu